thirdparty/runtime_environments.go

```diff
@@ -0,0 +1,51 @@
+package thirdparty
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+type RuntimeEnvironmentsClient struct {
+	Client  *http.Client
+	BaseURL string
+	APIKey  string
+}
+
+// getImageNames returns a list of strings containing the names of all images from the runtime environments api
+// TODO: Remove nolint:unused when DEVPROD-6983 is resolved.
+//
+//nolint:unused
+func getImageNames(ctx context.Context, c *RuntimeEnvironmentsClient) ([]string, error) {
+	apiURL := fmt.Sprintf("%s/rest/api/v1/imageList", c.BaseURL)
+	request, err := http.NewRequest(http.MethodGet, apiURL, nil)
+	if err != nil {
+		return nil, err
+	}
+	request = request.WithContext(ctx)
+	request.Header.Add("Content-Type", "application/json")
+	request.Header.Add("api-key", c.APIKey)
+	result, err := c.Client.Do(request)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	if result.StatusCode != http.StatusOK {
+		msg, _ := io.ReadAll(result.Body)
+		return nil, errors.Errorf("HTTP request returned unexpected status `%v`: %v", result.Status, string(msg))
+	}
+	var imageList []string
+	if err := json.NewDecoder(result.Body).Decode(&imageList); err != nil {
+		return nil, errors.Wrap(err, "Unable to decode http body")
+	}
+	var filteredImageList []string // filter out empty values
```

Comment:
nit 1: It's pretty clear just from L44-49 that it's filtering out empty values, so there's no need for the comment. Comments in general aren't necessary if they just reiterate what the code is doing. There's some info in [our style guide](https://docs.google.com/document/d/1-DB1FQ_zb5fX_0pnVcwyU0HmJPrloFBu6qSE8dkk8PU/edit#heading=h.l0w52tj2sr14) about when to use a comment.

nit 2: re the comment above about variable naming, `filteredImages` is more idiomatic.